### DISCUSSION TITLE
Value Formatting in Legend

### DISCUSSION THREAD

**Discussion by JBirtQueen** [eclipse-birt/birt#934] [Q&A] [answered]
I currently have a chart with Title X, Value 600, Title Y Value 700. I would like the legend to read Title X Value 600.00, Title Y Value 700.00, but can only find details on how to make values show as percent in legend. I have tried updating the formatting on the axis, replicating a script based on this eclipse thread [https://www.eclipse.org/forums/index.php/t/204633/](url) and formatted the value in the data set. I'm guessing I need a script, but I cannot figure out what elements I need to call out for it.

**Comment by JBirtQueen** [comment]
This is my script function beforeDrawLegendItem( lerh, bounds, icsc )
{
var curr = lerh.getValueLabel().getCaption().getValue();
nf = Formatter.format(lerh.getValueLabel().getCaption().getValue().replaceAll("600","700"),"###,###.00")
str = nf
lerh.getValueLabel().getCaption().setValue(str);
}

What comes up with this is 600###,###.00 which is not what I am looking for. I feel like I am very close to getting to the right thing for this...

**Reply by claesrosell** [reply]
Does this work?

```
function beforeDrawLegendItem( lerh, bounds, icsc )
{
     var curr = lerh.getValueLabel().getCaption().getValue();
     var str = Number(curr).toFixed(2);
     lerh.getValueLabel().getCaption().setValue(str);
}
```

**Reply by JBirtQueen** [reply]
Close - but I get NaN as my result. So I tried function beforeDrawLegendItem( lerh, bounds, icsc )
{
var curr = lerh.getValueLabel().getCaption().getValue();
var str = parseInt(curr,10);
lerh.getValueLabel().getCaption().setValue(str);
}

Which of course only gives the first number it encounters as a whole integer; in my case for 600/700 it returns the whole value, but if the legend value shows 10,000, it returns 10, which says to me the whole thing is already a string value. So perhaps I should count characters from the end of the value and if it is no decimal append .00, if it is .#, append 0, and if it is .## append nothing? That might be easier

**Comment by JBirtQueen** [comment, marked as answer]
After an extensive amount of trial and error - here's what worked:

function beforeDrawLegendItem( lerh, bounds, icsc )
{
var curr = lerh.getValueLabel().getCaption().getValue()
var stri=curr.toString()
var str2=stri.replace(",","")
var nf = Number(str2).toFixed(2)
var nf2 = nf.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
var str = nf2
lerh.getValueLabel().getCaption().setValue(str);
	
}

The original value is a string object, so it has to be converted first to a string. Then replace the ',' because that stops it from becoming a number. Then put the "," back by using RegExp. 

Thank you @claesrosell for the assist!
